### ISSUE TITLE
[8.0.0+] value population from an edited cell results in a console error a population does not execute  - use of CMD + ENTER

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#7382]
### Description

Once we select a header and try to populate values from an edited cell we get a console error a population does not execute.

### Steps to reproduce

1. Select any header
2. Type 'www'
3. Use <kbd>CMD</kbd> + <kbd>ENTER</kbd>

**Result 7.4.2 and lower**
Value is populated vertically. There's no error in the console.

**Result for 8.0.0. and newer**
Value is not populated. We get a console error.
```
Uncaught Error: Assertion failed: Expecting an unsigned number.
    at assert (handsontable.full.js:20470)
    at LazyFactoryMap.obtain (handsontable.full.js:61270)
    at CellMeta.getMeta (handsontable.full.js:80928)
    at MetaManager.getCellMeta (handsontable.full.js:80235)
    at Core.getCellMeta (handsontable.full.js:55404)
    at Object.populateFromArray (handsontable.full.js:53392)
    at Core.populateFromArray (handsontable.full.js:54066)
    at TextEditor.saveValue (handsontable.full.js:18640)
    at TextEditor.finishEditing (handsontable.full.js:18735)
    at EditorManager.closeEditor (handsontable.full.js:73844)
```

### Demo

http://jsfiddle.net/8xtnyar6/2/ - 7.4.2 - works well
http://jsfiddle.net/8xtnyar6/3/ - 8.0.0 - broken

![cmd_enter_bug](https://user-images.githubusercontent.com/11849621/104593699-565a0c80-5670-11eb-9a53-d2fcc1474e32.gif)

### Your environment
* Handsontable version: 8.0.0+
* Browser Name and version: Chrome 86
* Operating System: macOS Catalina 10.15.7

Inform zen 25669

**Comment by AMBudnik** [comment]
I also need to add that when we click <kbd>CMD</kbd>+<kbd>Enter</kbd> made times we are not able to exit the cell editor but selection changes.

![bug](https://user-images.githubusercontent.com/10757813/99387305-0b23b780-28d4-11eb-8883-b27a252ee745.gif)

**Comment by aninde** [comment]
This issue has been fixed by v8.3.2 http://jsfiddle.net/ezphLky9/. There is no error in the console after population from an edited cell.

![Kapture 2021-03-10 at 16 40 07](https://user-images.githubusercontent.com/30186818/110655380-4cabdb80-81bf-11eb-8b74-f9bbc5984188.gif)

Tested on 
macOS 11.2.2 (20D80) - Chrome  `88.0.4324.192`, Safari  `14.0.3 (16610.4.3.1.4)`, Firefox `86.0` , Edge `89.0.774.48`, Opera `74.0.3911.203` 
Windows 10 - Chrome  `88.0.4324.192` , Edge `89.0.774.48`, IE `11`, Firefox `86.0`.

**Comment by AMBudnik** [comment]
Thank you guys, I've sent the update further.
